numpy/core/include/numpy/npy_3kcompat.h

```diff
@@ -221,33 +221,29 @@ npy_PyFile_Dup2(PyObject *file, char *mode, npy_off_t *orig_pos)
                         "Getting a FILE* from a Python file object failed");
     }
 
+    ret = PyObject_CallMethod(file, "seekable", NULL);
+    if (ret == NULL){
+        fclose(handle);
```

Comment:
Why not check for seekability before calling `fdopen`, so that you can avoid having to open then close the file if `seekable()` fails?

Reply:
Alright, I put it at the very top, even before flush. I think that makes sense.